### PR TITLE
fuzzer: fix CLang 16

### PR DESCRIPTION
**Pull request by JacekGlen**
- Update the `compiler_install.sh` script
- Update the job definition to ensure that the build config is always cleaned up
- Increase memory available to each fuzzer job

### CODE REVIEW

File: cmake/setup/compiler_install.sh [hunk 1]
```diff
@@ -17,15 +17,31 @@ function install_gcc {
...
-	sudo update-alternatives --install /usr/bin/c++ c++ /usr/bin/clang++ 100
-	sudo update-alternatives --install /usr/bin/llvm-cov llvm-cov /usr/bin/llvm-cov-$CLANG_VERSION 100
-	sudo update-alternatives --install /usr/bin/llvm-profdata llvm-profdata /usr/bin/llvm-profdata-$CLANG_VERSION 100
+	
+	if [[ "$CLANG_VERSION" -ge 16 ]]
+	then
+		# Clang 16 and later versions have a different package repository
+		# Get the latest installation script and execute
+ 		sudo rm -f llvm.sh
+ 		sudo wget https://apt.llvm.org/llvm.sh
```
Comment (battlmonstr) [inline cmake/setup/compiler_install.sh:27]: this shell script should be committed into third_party (as we do for wasmer) and reviewed

File: cmake/setup/compiler_install.sh [hunk 2]
```diff
@@ -17,15 +17,31 @@ function install_gcc {
...
+		sudo chmod u+x llvm.sh
+		sudo ./llvm.sh $CLANG_VERSION
+		sudo apt-get install -y libc++-$CLANG_VERSION-dev libc++abi-$CLANG_VERSION-dev
+		sudo ln -sfv /usr/bin/clang-$CLANG_VERSION /usr/bin/clang
+		sudo ln -sfv /usr/bin/clang++-$CLANG_VERSION /usr/bin/clang++
+		sudo update-alternatives --install /usr/bin/cc cc /usr/bin/clang 100
+		sudo update-alternatives --install /usr/bin/c++ c++ /usr/bin/clang++ 100
```
Comment (battlmonstr) [inline cmake/setup/compiler_install.sh:34]: this section is almost identical to the lines below, could we refactor it to not copy paste?

in particular, we need llvm-cov and llvm-profdata for some CI jobs

**Review by battlmonstr — CHANGES_REQUESTED** [review]
@JacekGlen this new shell script seems to support the old versions as well. Could we use it for everything?

File: cmake/setup/compiler_install.sh (hunk 1 above)
Comment (JacekGlen) [reply cmake/setup/compiler_install.sh]: Done

File: cmake/setup/compiler_install.sh (hunk 2 above)
Comment (JacekGlen) [reply cmake/setup/compiler_install.sh]: Done

**Comment by JacekGlen** [comment]
> @JacekGlen this new shell script seems to support the old versions as well. Could we use it for everything?

I think the new approach combines the best of two worlds: it is backwards compatible and uses llvm shell script for all versions.

The reason I didn't use the llvm script for everything is that their apt repository do not all the versions available for all the distributions. For example they only have CLang >= 17 for Ubuntu 24, see: https://apt.llvm.org/noble/dists/

**Review by battlmonstr — APPROVED** [review]

**Comment by JacekGlen** [comment]
Fixes #2230

**Pull request merged**